src/librustc_session/config.rs

```diff
@@ -701,6 +702,8 @@ pub enum CrateType {
     ProcMacro,
 }
 
+impl_stable_hash_via_hash!(CrateType);
```

Comment:
Why was this necessary?

Reply:
We're removing the HashStable derive as that depends on details specific to librustc (I think there's not currently a derive for HashStable that is via Hash).

Reply:
Ah; that's unfortunate -- perhaps leave a FIXME for now?

Reply:
There are ~20 uses; I personally consider this better than the proc macro (and would expect it to be faster in a compilation time sense, though could be wrong).

src/librustc_session/session.rs:2
src/librustc_session/search_paths.rs:1
src/librustc_session/lint.rs:1
src/librustc_session/config.rs:8
src/librustc_session/utils.rs:1
src/librustc_errors/lib.rs:1
src/librustc_data_structures/fingerprint.rs:1
src/librustc_data_structures/stable_hasher.rs:17